core/arch/x86/decode_table.c

```diff
@@ -1458,6 +1458,7 @@ const instr_info_t * const op_instr[] =
 #define H15_dq TYPE_H, OPSZ_15_of_16
 #define Hqq TYPE_H, OPSZ_32
 #define Hx TYPE_H, OPSZ_16_vex32
+#define Hvq_dq TYPE_H, OPSZ_8_of_16_vex32
```

Comment:
The code looks correct, but the commit description is confusing:

> DynamoRIO does not break down exactly the memory locations that are read by vunpck*
> instructions in its operand size type. The OPSZ_8_of_16_vex32 does not exactly apply
> to the vex version as actually only 16 bytes are read.

Remove "only": it is very confusing as it implies a smaller number and it makes it look like your "16" is a typo.  Your point is that for vex32, 16 of the 32 bytes are read, not 8 of the 32: basically it's reading half.

So for 64-byte zmm, it's 32_of_64.

Maybe we need a OPSZ_half_of_16_vex32, or else OPSZ_8_of_16_vex16_of_32 (plus whatever is needed for zmm).  File an issue?

Reply:
Done. Ok. I have a patch in the pipeline for instructions above in evex. I will incorporate that into the patch. 

OPSZ_8_of_16_vex16_of_32
OPSZ_8_of_16_vex16_of_32_evex32_of_64

I think this is better:

OPSZ_half_of_16_vex32
OPSZ_half_of_16_vex32_evex64

?

Reply:
Sure.